### ISSUE TITLE
Users without SUPER can define User Group Member for all users and all companies

### ISSUE THREAD

**Issue by marekjz** [microsoft/AL#6722]
BC handles security around managing Permission Sets, User Group Permission Sets, User Group Access Control and Access Control records quite well in many cases, giving the following validation error when a user without SUPER-permissions attempts to manage them:
_You have not been granted rights to add, modify, or delete this permission set for this user in this company. To obtain rights, you must be granted either the SUPER or SECURITY permission set._

But when it comes to managing User Group Members - there is still a security hole that allows malicious (or unexperienced) user to leverage permissions defined later by authorized administrator to grant permissions to arbitrary user in any company. Namely, any user with WRITE permission to User Group Member table can manage (add, edit) records in that table defined for any user in any company (!), at least as long there are no User Group Permission Sets defined for that group (if there are, then the above error message would pop up, as a result of triggered Access Control creation attempt).

User Group Permission Sets can be defined later than User Groups, which might give enough time to someone to assign unauthorized users to groups that will later get assigned permissions, without throwing any error, so when an administrator finally assigns User Group Permission Sets to that group, it will trigger assigning Access Control records also for those "unintended" users.

What I would expect is to:
- Get an error message when a user without SUPER-permissions attempts to manage User Group Members. Only WRITE permission should not be enough to manage User Group Members, for any user (including the user who is attempting to do that).
- Get an error message when a user with only per-company SUPER-permissions attempts to manage User Group Members not attached to companies the user has SUPER in (also those without specified company, i.e. "global").
- Allow only users with "global" SUPER-permissions to manage "global" User Group Members (i.e. without specified Company Name).
- Allow users with SUPER in "global" scope or scope of the company defined on specific User Group Member to manage that record.

Bug observed in: Version: NO Business Central **18.0** (Platform 18.0.22893.23773 + Application 18.0.23013.23795)

**Comment by marekjz** [comment]
There should also be some restrictions on creation of User Groups - user should be at least SUPER/SECURITY in context of some company to be able to manage (create, edit) User Groups. MAYBE user should be "global" SUPER/SECURITY to be able to do that.

**Comment by db-bc** [comment]
I don't see how this is relate to the AL language/compiler, but rather the platform.

**Comment by marekjz** [comment]
@db-bc That's right, probably as well as: https://github.com/microsoft/ALAppExtensions/issues/13556
So would you accept this ticket as "received", or should I send the information via another channel?

**Comment by marekjz** [comment]
@db-bc Anyway, let's see what Microsoft guys will say about it here :)

**Comment by thpeder** [comment]
Hi @marekjz,
Thanks a lot for reporting the issue.
We would like you to report security related issues to secure@microsoft.com and that includes this one.
